pgmpy/factors/CPD.py

```diff
@@ -107,27 +107,25 @@ def __init__(self, variable, variable_card, values,
 
         cardinality = [variable_card]
         if evidence_card is not None:
-            if not isinstance(evidence_card, (list, set, tuple)):
+            if not isinstance(evidence_card, (list, tuple)):
                 if isinstance(evidence_card, np.ndarray):
                     evidence_card = evidence_card.tolist()
                 elif isinstance(evidence_card, (int, float)):
                     evidence_card = [evidence_card]
                 else:
-                    raise TypeError("Evidence cardinality must be list, set, "
+                    raise TypeError("Evidence cardinality must be list, "
                                     "tuple, numpy ndarray or simply a number.")
-            self.evidence_card = evidence_card
             cardinality.extend(evidence_card[::-1])
```

Comment:
@anaviltripathi Actually we need to just make this point clear in our docs. For example with the implementation that you have done:
```python
 cpd = TabularCPD('A', 2, [[0.1, 0.2, 0.3, 0.4], [0.9, 0.8, 0.7, 0.6]], ['B', 'C'], [2, 2])
```
this results in a CPD like this:
```
╒═════╤═════╤═════╤═════╤═════╕
│ C   │ C_0 │ C_0 │ C_1 │ C_1 │
├─────┼─────┼─────┼─────┼─────┤
│ B   │ B_0 │ B_1 │ B_0 │ B_1 │
├─────┼─────┼─────┼─────┼─────┤
│ A_0 │ 0.1 │ 0.2 │ 0.3 │ 0.4 │
├─────┼─────┼─────┼─────┼─────┤
│ A_1 │ 0.9 │ 0.8 │ 0.7 │ 0.6 │
╘═════╧═════╧═════╧═════╧═════╛
```
And without the reversing it would have been:
```
╒═════╤═════╤═════╤═════╤═════╕
│ B   │ B_0 │ B_0 │ B_1 │ B_1 │
├─────┼─────┼─────┼─────┼─────┤
│  C   │ C_0 │ C_1 │ C_0 │ C_1 │
├─────┼─────┼─────┼─────┼─────┤
│ A_0 │ 0.1 │ 0.2 │ 0.3 │ 0.4 │
├─────┼─────┼─────┼─────┼─────┤
│ A_1 │ 0.9 │ 0.8 │ 0.7 │ 0.6 │
╘═════╧═════╧═════╧═════╧═════╛
```
So, we need to make this clear in the docs that how does the order of evidence affect the parents of the CPD.

And the previous behavior was the latter one. I think we should keep that only as that's more intuitive. What do you think ?